Assets/Countly/Plugins/CountlySDK/Services/UserDetailsCountlyService.cs

```diff
@@ -374,7 +470,25 @@ private void AddToCustomData(string key, object value)
             CustomDataProperties.Add(key, value);
         }
 
-        #region override Methods
+        /// <summary>
+        /// Checks if an internal call is providing valid key and has consent 
+        /// </summary>
+        /// <param name="key">property name</param>
+        private bool ValidateRequest(string key)
+        {
+            string caller = new System.Diagnostics.StackTrace().GetFrame(1).GetMethod().Name;
```

Comment:
look at this fancy pants. What will this print? How consistent it is? Just googling it gave me these:
-If compilation optimizations are enabled, the calling method may be inlined, so you may not get the value you expect. 
-async methods get compiled into a state machine, so again, you may not get what you expect.